Reworking compatibility issues with older devices

OverDriver05 opened the pull request:
Hello, the Serious Sam Android Ports are pretty great, but i can only play on Update 1.4.0 only, probably due to the latest update only working on newer devices.

I mean, the ports worked flawlessly on my Samsung Galaxy E7, so why does the newest update doesn't work?

Hopefully this would be fixed in the next update!

Skyrimus committed:
Some fixes

Skyrimus committed:
Something v2

Skyrimus committed:
Improve view models of weapons

Commit:
Fixed model view in Player Profile

Skyrimus committed:
Reborn ADVANCED OPTIONS menu

Skyrimus committed:
Quarter resolution :)

Skyrimus committed:
Implemented ip address in menu

Skyrimus committed:
Clear

Skyrimus committed:
Implemented ping in hud

Skyrimus committed:
Improve networking

Skyrimus committed:
Improv networking #2

Skyrimus committed:
Improve networking #3

Skyrimus committed:
OnTouch MoveEvent use switch/case

Skyrimus committed:
Mapping 1 try

Skyrimus committed:
Ugly buttons mapping implementation

Implemented: c
1) Change buttons positions
2) Save positions after game restart

Skyrimus committed:
Ugly buttons mapping implementation fix

Skyrimus committed:
Update MainActivity.java

Skyrimus committed:
Buttons mapping dynamically position update

Skyrimus committed:
MotionEvent switch/case finally

Skyrimus committed:
Improve touch tracking

Skyrimus committed:
Buttons mapping mode activate/deactivate

To activate/deactivate "Buttons mapping mode" if long press Menu button in game

First long press on Menu button -
Buttons mapping: ON
Second long press on Menu button -
Buttons mapping: OFF

Skyrimus committed:
Joystick improvement

Skyrimus committed:
"Apply" button for "Buttons mapping mode"

Skyrimus committed:
Try to buttons size change

Skyrimus committed:
SeriousBomb button visible for "Buttons mapping mode"

Skyrimus committed:
Remap default buttons positions

Skyrimus committed:
Update MainActivity.java

Skyrimus committed:
Reborn "AUDIO OPTIONS" menu

Now you can adjust volume of in-game sound effects and in-game music

Skyrimus committed:
Update ButtonView.java

Skyrimus committed:
Classic green UI for TFE levels

Skyrimus committed:
TFE HUD 100% as PC version

Skyrimus committed:
Some infos saves in log

Skyrimus committed:
More info

Skyrimus committed:
App name is Serious Sam

Skyrimus committed:
Sort filenames in CMakeLists.txt

Skyrimus committed:
TFE Implementation (beta)

Now TFE can start game

Need to place SE1_10.gro in SeriousSam folder with TFE *.gro files and Levels folder

Skyrimus committed:
TFE Fixes 1

Skyrimus committed:
Include config.h for TFE

Skyrimus committed:
Update SeriousSam.cpp

Skyrimus committed:
TFE Fixes 2

Skyrimus committed:
Networking improvements

Skyrimus committed:
Networking improvements

Skyrimus committed:
Revert "Improve networking #3"

This reverts commit b492d8c0dbfc54bbb7cde049d02d4203dbc50c12.

Skyrimus committed:
Revert "Improve networking"

This reverts commit 56a3d31793e116acf0a1a3f3cd23803c3ee38f2d.

Skyrimus committed:
Update CMakeLists.txt

Skyrimus committed:
TFE Fixed enemy portal effect

Skyrimus committed:
Update WeaponPositions.h

Skyrimus committed:
Enable/Disable banner in right top of screen

Skyrimus committed:
Detect wlan0/ap0 interface for IP in menu

Skyrimus committed:
LimitFrameRate enabled

Skyrimus committed:
Merge branch 'master' into master

Skyrimus committed:
Update SeriousSam.cpp

Skyrimus committed:
Fix Armor in HUD

Skyrimus committed:
Fix Armor in HUD

Skyrimus committed:
TFE fixes and some things

Skyrimus committed:
Build apks for TSE and TFE at the same time

Now we can play in TSE and TFE from different apks
1) Now game files for TSE locate in SeriousSamTSE folder
2) Now game files for TFE locate in SeriousSamTFE folder
3) Added new icon and splash for TFE launcher

Move your TSE game files from /sdcard/SeriousSam/ to
/sdcard/SeriousSamTSE/

To start TFE you need copy TFE game files to /sdcard/SeriousSamTFE/ folder

Game files for TFE from Steam:

* Levels (folder)
* 1_00_ExtraTools.gro
* 1_00_music.gro
* 1_00c.gro
* 1_00c_Logo.gro
* 1_00c_scripts.gro
* 1_04_patch.gro

Skyrimus committed:
README for The First Encounter

Skyrimus committed:
README for The First Encounter

Skyrimus committed:
Update build.gradle

Skyrimus committed:
[Buttons mapping] Movable joystick

Now you can move joystick like any button on screen

Skyrimus committed:
Merge branch 'master' of https://github.com/Skyrimus/Serious-Sam-Android

Skyrimus committed:
[Buttons mapping] Changeable size of buttons

Added support for change size of buttons
Rework some features of buttons mapping
Clean code and improvements

Skyrimus committed:
[Buttons mapping] Clean code

Skyrimus committed:
CRC Check ok for all, badsyncs gone?

Need more tests
No one badsync for 40min game, hmm

Skyrimus committed:
Spawnflags same on different devices

Skyrimus committed:
Disable LimitFrameRate

Skyrimus committed:
Fixes (CRC, Player Personalize in single player, Highscore recording)

Force set CRC for some files
Unlocked player and controls menu in Single Player game
Enabled highscore saving (need fix loading after restart)

Skyrimus committed:
Some fixes

Gyroscope fix (disabled by default)
INT to SINT
Some misc fixes

Skyrimus committed:
GyroSensibility fix

Skyrimus committed:
Fully fixed game translations

Now all lines in the game are can translated into another language

Skyrimus committed:
Update MAudioOptions.cpp

Skyrimus committed:
Some fixes

Skyrimus committed:
Setup arm64-v8a build

Skyrimus committed:
Some void fixes

Skyrimus committed:
Fixes (cosnst char, indexes,and etc)

Skyrimus committed:
Engine network buffer include

Skyrimus committed:
arm64-v8a build fixes

Skyrimus committed:
Fixes (etc)

Skyrimus committed:
Fixes (etc)

sergioroprado committed:
Update build.gradle

Replaced the dead Jcenter/Bintray (to fix the compile problem)

sergioroprado committed:
Update build.gradle

Since this project haven't migrated to AndroidX, the KeyboardHeightProvider dependency had to be replaced with another version without AndroidX, to avoid compilation errors.

Skyrimus committed:
Update Texture.cpp

Skyrimus committed:
Merge pull request #1 from sergioroprado/patch-1

Update build.gradle

Skyrimus committed:
Merge pull request #2 from sergioroprado/patch-2

Update build.gradle

Skyrimus committed:
Android TV Support thx sergioprado

Skyrimus committed:
Debug build apk variant works

Skyrimus committed:
Some sound fixes

Skyrimus committed:
Some hueta blyat

Skyrimus committed:
Some hueta opyat

Skyrimus committed:
Somesome

Skyrimus committed:
Some fixes (etc and more)

Skyrimus committed:
Update README.md

Skyrimus committed:
Fix weapon positions

Skyrimus committed:
Some arm fixes

Skyrimus committed:
Some arm fixes

Skyrimus committed:
More fixes

Player states (swim, cround and etc)
Enemies projectiles fixes
Fix player speed without weapon in DM
Some network fixes from SSE

Skyrimus committed:
New NetSettings scripts, fixed inf ammo cheat and more

More network fixes and tweaks from SSE

Skyrimus committed:
Revert some SSE changes

Skyrimus committed:
Update Network.cpp

Skyrimus committed:
Install apk on multiple devices

Can use in windows if your OS containts git application

Skyrimus committed:
Minor changes

Skyrimus committed:
Servers browser ported!

1) Added support for searching servers
2) Added support for DarkPlaces Master Server
3) Now we can search servers in LAN network and Internet, BUT because port based on Serious Engine 1.10 you can connect to 1.10 android servers ONLY.
4) Now servers browser use master server from 1.07 engine version, later I will try to setup own DarkPlaces Master Server for Android servers.
5) Search Internet button disabled now while I don't setup new master server.

Skyrimus committed:
Revert some changes from SSE

Skyrimus committed:
Update Network.cpp

Skyrimus committed:
Revert "Update Network.cpp"

This reverts commit 29c857ae1751189932524c3f2c3f4e72d2d4d6ae.

Skyrimus committed:
Update MGServerList.cpp

Skyrimus committed:
ServerList ui scaling

Skyrimus committed:
Some shit

Skyrimus committed:
Forgot add EnumCancel

Skyrimus committed:
Net fix

Skyrimus committed:
Socket fixes for server browser

Skyrimus committed:
Update MasterServerMgr.cpp

Skyrimus committed:
Advanced Monster & Item Classes mod implementation

Many thanks to Heming_Hitrowski for providing the mod sources

Skyrimus committed:
Separate addition jniLibs folders

Skyrimus committed:
Some changes

Skyrimus committed:
Delete useless source code

Skyrimus committed:
Some

Skyrimus committed:
Desync on arm/arm64 devices fully fixed

Skyrimus committed:
Some minor changes

Skyrimus committed:
Stream changes

Skyrimus committed:
Stream changes 2

Skyrimus committed:
It seems better

Skyrimus committed:
arm64-v8a build fixes

Skyrimus committed:
Use libtremor for ogg vorbis decoding

Skyrimus committed:
TFE Minigun sound offsets bug fixed

Skyrimus committed:
Lights and strings fixes

Skyrimus committed:
armeabi-v7a and arm64-v8a fixes

Skyrimus committed:
Update GfxLibrary.cpp

Skyrimus committed:
Update GfxLibrary.cpp

Skyrimus committed:
Timer low precission added

Skyrimus committed:
Update libEntitiesAdvMP.so

Skyrimus committed:
arm64-v8a fully implemented, added Advanced Monster & Item Classes mod for arm64-v8a variant

Skyrimus committed:
H3D mod implementation

Skyrimus committed:
Safe start in user copied all files and folders from Steam game

Skyrimus committed:
Update Stream.cpp

Skyrimus committed:
Cosmetical changes

Skyrimus committed:
Update SeriousSam.cpp

Skyrimus committed:
Don't render crosshair if player dead

Skyrimus committed:
Some byteswap

Skyrimus committed:
Fixed model rendering in NETRICSA

Skyrimus committed:
Some fixes

Skyrimus committed:
Update version.properties

Skyrimus committed:
Update Woman.es

Skyrimus committed:
Migrated to AndroidX

Skyrimus committed:
Update Color.h

Skyrimus committed:
Update Color.cpp

Skyrimus committed:
Update Color.h

Skyrimus committed:
Update Color.h

Skyrimus committed:
Update Color.cpp

Skyrimus committed:
Update Vertex.h

Skyrimus committed:
Update README.md

Skyrimus committed:
Update README.md

Skyrimus committed:
[skip ci] Update appveyor.yml

Skyrimus committed:
ECL replacing file

Skyrimus committed:
Added zcaliptium to CONTRIBUTORS

Skyrimus committed:
Some sounds improvement

Skyrimus committed:
Agly mouse control implementation

Scroll bug:
Change to prev/next weapon works if you scroll mouse wheel twice only, because if set bWeaponPrev or bWeaponNext to 1 and to 0 once, this is not works, used agly crutch

SeriousAlexej committed:
Draw crosshair over nearby enemy

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

Skyrimus committed:
Fix case in string in SessionState

SeriousAlexej committed:
Reuse ray radius as ratio coefficient

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Ignore players for autoaim

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Redirect weapon to autoaim target

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Add crosshair oscilation

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Proper initialization of rays

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Surround linkage to win-specific libs with ifdef

SeriousAlexej committed:
Rewrite NULL checks in paranoid mode

SeriousAlexej committed:
Use lerped autoaim pos for smooth crosshair rendering

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Make crosshair larger

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Remove old cvars, add new plr_fAutoAimSensitivity

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

Skyrimus committed:
Update CONTRIBUTORS.txt

SeriousAlexej committed:
Render autoaim target separately from crosshair

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Include auto aim texture into the engine

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

Skyrimus committed:
Merge pull request #3 from SeriousAlexej/autoaim

Implementation of auto-aim to greatly increase playability

Skyrimus committed:
TFE Auto aim support

Skyrimus committed:
aim turns on 0.1 sensitivity

Skyrimus committed:
Aim Assistant settings in "Android Port Settings" menu

Skyrimus committed:
Update preferences.xml

Skyrimus committed:
Update prebuilt H3D and Advanced Monster & Item Classes mods

Added aim-assistant support to H3D

Skyrimus committed:
[skip ci] Test

Skyrimus committed:
[skip ci] Revert

Skyrimus committed:
Update gradle to 7.0.3

Skyrimus committed:
[skip ci] Some shit

Skyrimus committed:
[skip ci]

Skyrimus committed:
Create update.json

Skyrimus committed:
Update update.json

Skyrimus committed:
[skip ci]

Skyrimus committed:
[skip ci]

Skyrimus committed:
[skip ci]

Skyrimus committed:
Apk auto update implemented

Fixed IP address getting in menu

Skyrimus committed:
Merge branch 'master' of https://github.com/Skyrimus/Serious-Sam-Android

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Russian translation in Android Port Settings

Skyrimus committed:
[skip ci]

Skyrimus committed:
Update version.properties

Skyrimus committed:
Revert "Update gradle to 7.0.3"

This reverts commit f0828f6bf5589ed3f0f5d70493c2c7c9fcec76b1.

Skyrimus committed:
Update MainActivity.java

Skyrimus committed:
1.05.2

Skyrimus committed:
Revert "Update appveyor.yml"

This reverts commit 0f543feea7414736e4929277b04e326b5dd8c6e1.

SeriousAlexej committed:
Add ability to reload in-memory resources

Signed-off-by: SeriousAlexej <seriousalexej@gmail.com>

SeriousAlexej committed:
Merge remote-tracking branch 'skyrimus/master' into autoaim

SeriousAlexej committed:
Same fix for TFE

SeriousAlexej committed:
Undo whitespace change

SeriousAlexej committed:
Fix endif without comment

Skyrimus committed:
Merge pull request #4 from SeriousAlexej/autoaim

Add ability to reload in-memory resources

Skyrimus committed:
Minor fixes

Skyrimus committed:
Buttons Mapping Rework Part 1

Skyrimus committed:
Buttons Mapping Rework Part 2 + fixes

H3D updated to v97
\t fixed in java files
Moved g_IncomingControls in binding callbacks, setted g_incomingControls as current controls globally
Added support to load custom GameMP libraries

Skyrimus committed:
Update playerCommons.h

Skyrimus committed:
Update playerCommons.h

Skyrimus committed:
Controls fixes

Skyrimus committed:
Buttons Mapping Rework Part 3

Skyrimus committed:
Minor changes

Skyrimus committed:
TFE launch fixed

Skyrimus committed:
Don't generate SeriousBomb button in TFE

Skyrimus committed:
Volume keys actions implemented

Skyrimus committed:
Revert ugly mouse implementation

This have crash on some devices

Skyrimus commented:
@OverDriver05 Android 4.4 Kitkat support already deprecated for this app, compiling apk with elder sdk api not good idea, because now project rebased to androidx repo. Your solution - buying new phone with latest Android OS. Please close pull request and have fun :)

Skyrimus committed:
Navigation button "Back" fix for NETRICSA

Skyrimus committed:
TFE Green color text

Skyrimus committed:
Light fix by @zcaliptium

Skyrimus committed:
Fixed C++ DitherBitmap function by Tin Kesaro

Skyrimus committed:
v1.05.3

Skyrimus committed:
GLES 3.0 tries

Skyrimus committed:
Refactoring Part 1

1) Added exception catch for translation loading
2) Hiding touch controls now set alpha transpartncy for buttons backgroud and not delete buttons from layout
3) All on screen buttons now have same color (looks better)
4) Smooth and relative mouse options forced to False

Skyrimus committed:
Refactoring Part 2

1) Updater moved in separated class
2) ButtonSet moved in separated class
3) Rewrited getWifiIP function

Skyrimus committed:
Build improvements

1) Build logs separated to logs folder
2) Logs name now if datetime of build started
3) Added custom add_subdirectory implementation with checking existing subdirectory
4) Deleted mfpu flag in compileflags
5) Added gitignore for H3D folder in Sources and logs folder in root
6) Separated Advanced Monster & Items Classes mod and H3D to different folders
7) Added checks for existing H3D sources, if H3D sources not finded - copying prebuilt H3D libs from libs_tse_h3d folder

Skyrimus committed:
Build improvements 2

Added support for add compiled H3D libraries to prebuilt libs folder

Skyrimus committed:
Added copyFile func

Skyrimus committed:
Cache download & unpack implementation

* Implemented cache downloader for TFE and TSE
* Implemented cache unpacker
* Implemented UI for cache downloader an unpacker
* Deleted gradle logger :(

Skyrimus committed:
Gradle, Java and etc upstream

Upstreamed:
* JDK to JDK 17
* Gradle to 7.2 and AGL 7.0.2
* Gson to 2.8.6
* FileDownloader to 1.7.7

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update appveyor.yml

Skyrimus committed:
Update todo.txt

Skyrimus committed:
Some shit from last year

Skyrimus committed:
:)

Skyrimus committed:
Update gradle and agp

Skyrimus committed:
Restored default axis smoothing

Skyrimus committed:
Experimental touch camera improvement

Skyrimus committed:
Determinate FPU (need more tests)

Skyrimus committed:
GLES Adapter optimizations

Skyrimus committed:
Android dedicated server for Android arm64 (needed STATIC_LINKING ON + gamefiles)

Workdir  /data/local/tmp/SeriousSamTSE/
Copy game files and Scripts to workdir
Copy built static DedicatedServerAndroid from apk intermediates to workdir
chmod +x DedicatedServerAndroid
./DedicatedServerAndroid DefaultCoop

Skyrimus committed:
Make Dedicated Server compile on amd64

Skyrimus committed:
Full gamepad support (works in menu + netricsa) + Gamepad bindings can be changed + fixed Cache Downloader + renew Updater + mouse support (wheel not working)

Skyrimus committed:
Revert "Full gamepad support (works in menu + netricsa) + Gamepad bindings can be changed + fixed Cache Downloader + renew Updater + mouse support (wheel not working)"

This reverts commit 1819157d0a78010f0f02251f4b6f6549e8bdfbfc.
